src/toolbarActions/toolbarFunctions.js

```diff
@@ -80,15 +80,66 @@ const saveAction = (state) => {
 };
 
 async function saveGraphMLFile(state) {
+    const pickerOpts = {
+        types: [
+            {
+                description: 'Graphml',
+                accept: {
+                    'text/graphml': ['.graphml'],
+                },
+            },
+            {
+                description: 'Python',
+                accept: {
+                    'text/python': ['.py'],
+                },
+            },
+            {
+                description: 'C++',
+                accept: {
+                    'text/C++': ['.cpp'],
+                },
+            },
+            {
+                description: 'Shell scipt',
+                accept: {
+                    'text/script': ['.sh'],
+                },
+            },
+            {
+                description: 'V files',
+                accept: {
+                    'text/Vfiles': ['.v'],
+                },
+            },
+            {
+                description: 'C',
+                accept: {
+                    'text/C': ['.c'],
+                },
+            },
+            {
+                description: 'm',
+                accept: {
+                    'text/m': ['.m'],
+                },
+            },
+        ],
+        excludeAcceptAllOption: true,
+        multiple: false,
+    };
     if (state.curGraphInstance) {
         const graph = state.graphs[state.curGraphIndex];
-        if (graph.fileHandle) {
+        if (!graph.fileHandle) {
+            const fileHandle = await window.showSaveFilePicker(pickerOpts);
+            graph.fileHandle = fileHandle;
+        }
+        try {
             const stream = await graph.fileHandle.createWritable();
             await stream.write(getGraphFun(state).saveToFolder());
             await stream.close();
-        } else {
-            // eslint-disable-next-line no-alert
-            alert('Switch to Edge/Chrome!');
+        } catch (error) {
+            alert('Some error occured try again');
```

Comment:
No, this is intentional. This "Save" feature is supported only in Chrome.

Reply:
Concore app is not able to save unsaved files when clicked on "Save" 

What I have done is:
1)If that file is existing in pc, and open through "open" 
     a)If clicked on "Save" it will not ask for its locations and directly Save
2) If file is newly created and 
     a) if Clicked on "Save" It will ask for its location in PC to be saved and save , yeah I understand that it should Save only graphml files , so now I have removed that portion

What is happning before:
1)If that file is existing in pc, and open through "open" 
    a)If clicked on "Save" it will not ask for its location and directly Save
2)If file is newly created and
      a) If clicked on "Save" It will show alert and not able to Save the file  e.g   figure
         
![image](https://user-images.githubusercontent.com/100110395/217904354-d96180f6-fe4a-4c31-a793-c0aac284e18f.png)

I solved this issue , yeah I understand that It should accept only graphml files , I am sorry for that Now I removed that code of accepting different files

Reply:
Which browser you are using? As of now, only Chrome is able to save the file.

If your fix makes studies and programs saveable even in Firefox, it is a fix for a different issue (that is, "Enable Save studies and programs on Firefox").